src/tx-format/upgrade_purpose.md

```diff
@@ -0,0 +1,38 @@
+# UpgradePurposeType
+
+```c++
+enum UpgradePurposeType : uint8 {
+    ConsensusParameters = 0,
+    StateTransition = 1,
+}
+```
+
+| name   | type                                                                                        | description              |
+|--------|---------------------------------------------------------------------------------------------|--------------------------|
+| `type` | [`UpgradePurposeType`](#upgradepurposetype)                                                 | Type of upgrade purpose. |
+| `data` | One of [`ConsensusParameters`](#consensusparameters), [`StateTransition`](#statetransition) | Upgrade purposes.        |
+
+Transaction is invalid if:
+
+- `type > UpgradePurposeType.StateTransition`
```

Comment:
nit: I'd prefer to just say that "`type` is not valid `UpgradePurposeType` value"